### PR TITLE
Add blanks card

### PR DESCRIPTION
**Pull request by shawnclq**

### CODE REVIEW

File: src/main/java/seedu/address/storage/XmlAdaptedCard.java [hunk 1]
```diff
@@ -71,6 +74,11 @@ public Card toModelType() throws IllegalValueException {
         if (!Card.isValidCard(this.back)) {
             throw new IllegalValueException(Card.MESSAGE_CARD_CONSTRAINTS);
         }
+        if (FillBlanksCard.containsBlanks(this.front)) {
```
Comment (jethrokuan) [inline src/main/java/seedu/address/storage/XmlAdaptedCard.java]: this looks to be very similar to a normal card. The only reason I can think of for having a special card for this is when we want to reveal the blanks one at a time.

File: src/test/java/seedu/address/model/card/FillBlanksCardTest.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        Assert.assertThrows(NullPointerException.class, () ->
+                new FillBlanksCard(VALID_FILLBLANK_FRONT_SET, null));
+    }
+
+    @Test
+    public void constructor_invalidParam_throwsIllegalArgumentException() {
+        String invalidParam = " ";
+        List<String> invalidFrontSet = Arrays.asList(new String[] {invalidParam, "Hello", "World"});
+        List<String> invalidBackSet = Arrays.asList(new String[] {invalidParam, "Hello"});
+        Assert.assertThrows(IllegalArgumentException.class, () -> new FillBlanksCard(UUID.randomUUID(),
```
Comment (jethrokuan) [inline src/test/java/seedu/address/model/card/FillBlanksCardTest.java]: Create a constructor for FillBlanksCard that accepts fewer arguments and instantiates a UUID by default.

File: src/main/java/seedu/address/storage/XmlAdaptedCard.java (hunk 1 above)
Comment (shawnclq) [reply src/main/java/seedu/address/storage/XmlAdaptedCard.java]: You mean the implementation or you mean there's no need for fill-blanks card?

Comment (jethrokuan) [reply src/main/java/seedu/address/storage/XmlAdaptedCard.java]: Both. The similarities mean that there's possibility no need for special handling to get this kind of card differentiation. e.g. you could denote _ to be blanks and there would be no need for a new parser which supports multiple fronts and backs.

File: src/main/java/seedu/address/model/card/FillBlanksCard.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+//@@author shawnclq
+/**
+ * Represents a fill-in-the-blanks Flashcard.
+ * Guarantees: Front, Back must not be null.
+ *
+ */
+public class FillBlanksCard extends Card {
+
+    public static final String MESSAGE_FILLBLANKS_CARD_CONSTRAINTS =
```
Comment (jethrokuan) [inline src/main/java/seedu/address/model/card/FillBlanksCard.java]: this probably has to change

File: src/test/java/seedu/address/logic/parser/ParserUtilTest.java [hunk 4]
```diff
@@ -30,8 +30,14 @@
     private static final String VALID_MCQ_BACK = "2";
     private static final String VALID_MCQ_FRONT = "Which continent is Singapore in?";
     private static final String VALID_NAME = "Math";
-    private static final Set<String> VALID_MCQ_OPTIONS = new HashSet<>(
-            Arrays.asList(new String[]{"Australia", "Asia", "Africa"}));
+    private static final List<String> VALID_MCQ_OPTIONS =
```
Comment (jethrokuan) [inline src/test/java/seedu/address/logic/parser/ParserUtilTest.java]: `Arrays.asList("Australia", "Asia", "Africa")` is simpler, but either is fine.

File: src/test/java/seedu/address/logic/parser/ParserUtilTest.java [hunk 5]
```diff
@@ -159,13 +165,35 @@ public void parseMcqCard_validValue_returnsMcqCard() throws Exception {
         assertEquals(expectedMcqCard, ParserUtil.parseMcqCard(VALID_MCQ_FRONT, VALID_MCQ_BACK, VALID_MCQ_OPTIONS));
     }
 
+
+    @Test
+    public void parseFillBlanksCard_null_throwsNullPointerException() {
+        Assert.assertThrows(NullPointerException.class, () -> ParserUtil.parseFillBlanksCard((String) null,
+                VALID_FILLBLANKS_BACK));
+        Assert.assertThrows(NullPointerException.class, () -> ParserUtil.parseFillBlanksCard(VALID_FILLBLANKS_FRONT,
```
Comment (jethrokuan) [inline src/test/java/seedu/address/logic/parser/ParserUtilTest.java]: no need to cast null to `String` here.

**Pull request merged**